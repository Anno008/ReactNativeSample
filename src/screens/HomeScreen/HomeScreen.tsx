import * as React from "react";
import { Platform, Text, View } from "react-native";
import Todo from "../../components/todo";
import styles from "./HomeScreenStyles";

const instructions: string = Platform.select({
  android: `Android`,
  ios: `iOS`
});

export const HomeScreen: React.StatelessComponent = (): JSX.Element => (
  <View style={styles.container}>
    <Text style={styles.instructions}>Home screen</Text>
    <Text style={styles.instructions}>{instructions}</Text>
    <Todo />
  </View>
);

export default HomeScreen;
