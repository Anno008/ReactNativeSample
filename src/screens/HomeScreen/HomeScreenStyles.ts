import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: "#F5FCFF",
    flex: 1,
    justifyContent: "center",
  },
  instructions: {
    color: "#333333",
    marginBottom: 5,
    textAlign: "center",
  },
  welcome: {
    fontSize: 20,
    margin: 10,
    textAlign: "center",
  },
});
