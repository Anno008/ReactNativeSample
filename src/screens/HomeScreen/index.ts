export { default } from "screens/HomeScreen/HomeScreen";
