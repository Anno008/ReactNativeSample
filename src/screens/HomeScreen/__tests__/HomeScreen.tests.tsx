import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import { HomeScreen } from "../HomeScreen";

describe("HomeScreen test", () => {
  it("should render", () => {
    const wrapper: ShallowWrapper = shallow(<HomeScreen />);
    expect(wrapper).toMatchSnapshot();
  });
});
