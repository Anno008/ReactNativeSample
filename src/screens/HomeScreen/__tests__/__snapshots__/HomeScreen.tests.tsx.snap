// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen test should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          style={
            Object {
              "color": "#333333",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginBottom": 5,
              "textAlign": "center",
            }
          }
        >
          Home screen
        </Text>,
        <Text
          style={
            Object {
              "color": "#333333",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginBottom": 5,
              "textAlign": "center",
            }
          }
        >
          iOS
        </Text>,
        <Connect(Todo) />,
      ],
      "style": Object {
        "backgroundColor": "#F5FCFF",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Home screen",
          "style": Object {
            "color": "#333333",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 5,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Home screen",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "iOS",
          "style": Object {
            "color": "#333333",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 5,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "iOS",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            style={
              Object {
                "color": "#333333",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginBottom": 5,
                "textAlign": "center",
              }
            }
          >
            Home screen
          </Text>,
          <Text
            style={
              Object {
                "color": "#333333",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginBottom": 5,
                "textAlign": "center",
              }
            }
          >
            iOS
          </Text>,
          <Connect(Todo) />,
        ],
        "style": Object {
          "backgroundColor": "#F5FCFF",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Home screen",
            "style": Object {
              "color": "#333333",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginBottom": 5,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Home screen",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "iOS",
            "style": Object {
              "color": "#333333",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginBottom": 5,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "iOS",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
